### PR TITLE
CIV-6861 Display date field for Unless order

### PR DESCRIPTION
**Pull request by karthick-mohanasundaram-HMCTS**
**Before creating a pull request make sure that:**

- [x] commit messages are meaningful and follow good commit message guidelines
- [x] README and other documentation has been updated / added (if needed)
- [x] tests have been updated / new tests has been added (if needed)
- [x] Screenshot attached for the Evidence of manual testing
- [ ] Functional, Smoke and API tests have been run locally
- [x] Full gradle build and run tests with coverage have been completed (if needed)

Please remove this line and everything above and fill the following sections:

### Screen Shot ###

Refer attached word file

### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/CIV-6861

### Change description ###
Display optional date field for unless order when judge approve the order

Add Unless order scheduler

### Testing branch and instruction ###
GA and GA CCD
CIV-6861

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

**Comment by karthick-mohanasundaram-HMCTS** [comment]
image

File: Jenkinsfile_CNP [hunk 1]
```diff
@@ -9,7 +9,7 @@ def type = "java"
 def product = "civil"
 def component = "general-applications"
 def camundaBranch = "master"
-def generalappCCDBranch = "master"
+def generalappCCDBranch = "CIV-6861"
```
Comment (karthick-mohanasundaram-HMCTS) [inline Jenkinsfile_CNP:12]: Revert after testing

**Comment by karthick-mohanasundaram-HMCTS** [comment]
image

**Comment by karthick-mohanasundaram-HMCTS** [comment]
image

**Comment by karthick-mohanasundaram-HMCTS** [comment]
[Unless Order scheduler.docx](https://github.com/hmcts/civil-general-applications/files/10699090/Unless.Order.scheduler.docx)

**Comment by karthick-mohanasundaram-HMCTS** [comment]
image

File: src/main/java/uk/gov/hmcts/reform/civil/controllers/testingsupport/TestingSupportController.java [hunk 2]
```diff
@@ -146,4 +148,22 @@ public ResponseEntity<String> getJudgeRevisitProcessEvent(@PathVariable("state")
         return new ResponseEntity<>(responseMsg, HttpStatus.OK);
     }
 
+    @GetMapping("/testing-support/unless-order/trigger-judge-revisit-process-event/{state}")
```
Comment (vasudevganesanhmcts) [inline src/main/java/uk/gov/hmcts/reform/civil/controllers/testingsupport/TestingSupportController.java:151]: where are you using this ? i cant see an api test for it in ga ccd or that needs to be added

File: src/main/java/uk/gov/hmcts/reform/civil/service/tasklisteners/GAUnlessOrderSchedulerExternalTaskListener.java [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+
+import org.camunda.bpm.client.ExternalTaskClient;
+import org.camunda.bpm.client.topic.TopicSubscriptionBuilder;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnExpression;
+import org.springframework.stereotype.Component;
+import uk.gov.hmcts.reform.civil.handler.tasks.CheckUnlessOrderDeadlineEndTaskHandler;
+
+@Component
+@ConditionalOnExpression("${response.deadline.check.event.emitter.enabled:true}")
```
Comment (vasudevganesanhmcts) [inline src/main/java/uk/gov/hmcts/reform/civil/service/tasklisteners/GAUnlessOrderSchedulerExternalTaskListener.java:11]: this is not response deadline

File: src/main/java/uk/gov/hmcts/reform/civil/controllers/testingsupport/TestingSupportController.java (hunk 2 above)
Comment (karthick-mohanasundaram-HMCTS) [reply src/main/java/uk/gov/hmcts/reform/civil/controllers/testingsupport/TestingSupportController.java:151]: Hey Vasu, I have it to support api test.

File: src/main/java/uk/gov/hmcts/reform/civil/service/tasklisteners/GAUnlessOrderSchedulerExternalTaskListener.java (hunk 3 above)
Comment (karthick-mohanasundaram-HMCTS) [reply src/main/java/uk/gov/hmcts/reform/civil/service/tasklisteners/GAUnlessOrderSchedulerExternalTaskListener.java]: thanks for pointing it out.

File: src/main/java/uk/gov/hmcts/reform/civil/handler/tasks/CheckUnlessOrderDeadlineEndTaskHandler.java [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+    }
+
+    private List<CaseData> getUnlessOrderCasesThatAreEndingToday() {
+        List<CaseDetails> unlessOrderCases = caseSearchService
+            .getOrderMadeGeneralApplications(ORDER_MADE, UNLESS_ORDER);
+        return unlessOrderCases.stream()
+            .map(caseDetailsConverter::toCaseData)
+            .filter(caseData -> caseData.getJudicialDecisionMakeOrder()
+                .getJudgeApproveEditOptionDateForUnlessOrder() != null
+                && caseData.getGeneralAppType().getTypes().contains(UNLESS_ORDER)
```
Comment (Azam-Hmcts) [inline src/main/java/uk/gov/hmcts/reform/civil/handler/tasks/CheckUnlessOrderDeadlineEndTaskHandler.java:54]: We already filter for Apptype when we call cacheService at line 49, may be we can delete this condition as all application will be of type "UNLESS_ORDER"

File: src/main/java/uk/gov/hmcts/reform/civil/handler/tasks/CheckUnlessOrderDeadlineEndTaskHandler.java [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+
+import static java.time.LocalDate.now;
+import static uk.gov.hmcts.reform.civil.callback.CaseEvent.END_SCHEDULER_CHECK_UNLESS_ORDER_DEADLINE;
+import static uk.gov.hmcts.reform.civil.enums.CaseState.ORDER_MADE;
+import static uk.gov.hmcts.reform.civil.enums.dq.GeneralApplicationTypes.UNLESS_ORDER;
+
+@Slf4j
+@RequiredArgsConstructor
+@Component
+@ConditionalOnExpression("${judge.revisit.stayOrder.event.emitter.enabled:true}")
```
Comment (deepthidoppalapudihmcts) [inline src/main/java/uk/gov/hmcts/reform/civil/handler/tasks/CheckUnlessOrderDeadlineEndTaskHandler.java:29]: @karthick, we need to change this expression for unless order

**Review by vasudevganesanhmcts — APPROVED** [review]

**Review by Azam-Hmcts — APPROVED** [review]

**Pull request merged**